netlogo-gui/src/main/editor/CodeCompletionPopup.scala

```diff
@@ -0,0 +1,135 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.editor
+
+import java.awt.{Color, Component, Dimension}
+import java.awt.event._
+import javax.swing._
+
+import org.nlogo.api.CompilerServices
+import org.nlogo.core.DefaultTokenMapper
+import org.nlogo.window.AutoSuggest
+
+case class CodeCompletionPopup(compiler: CompilerServices) {
+  // To control when the popup has to automatically show
+  var isPopupEnabled = false
+  // To store the last value user selected
+  var lastSuggested = ""
+  var editorArea: EditorArea = null
```

Comment:
This should be `Option[EditorArea]` instead of `null`. In scala you should use an `Option` whenever you have a `var` that might be empty.